### PR TITLE
Fix default cache on windows

### PR DESCRIPTION
**Pull request by thomwolf**
Hard-coding the default cache lead to a broken path with slash on Windows

Fix https://github.com/huggingface/simenv/issues/217

### CODE REVIEW

File: src/huggingface_hub/constants.py [hunk 1]
```diff
@@ -70,9 +70,11 @@ def _is_true_or_auto(value: Optional[str]) -> bool:
 
 
 # default cache
+default_home = os.path.join("~", ".cache")
```
Comment (Wauplin) [inline src/huggingface_hub/constants.py:73]:
```suggestion
default_home = os.path.join(os.path.expanduser("~"), ".cache")
```

**Comment by Wauplin** [comment]
Thanks @thomwolf for the fix. I updated the code to expand `"~"` as well (it can have weird effect if `Path(...).resolve()` is used somewhere in the code). Will merge as soon as the CI passes :)

**Review by Wauplin — APPROVED** [review]

**Comment by thomwolf** [comment]
Awesome thanks, any idea of when the next release is expected to be?

**Comment by Wauplin** [comment]
@thomwolf is "soon" a good enough answer ? :D

(more pragmatically, I'd say most likely next week)

**Pull request merged**